### ISSUE TITLE
Unable to use keras.layers.Normalization on multiple GPUs

### ISSUE THREAD

**Issue by FalsoMoralista** [keras-team/keras#16059]
### Enviroment
I am using tensorflow 2.7.0 with python 3.8.10, linux cluster with multiple gpus.

### Context
I am trying to use a normalization layer as demonstrated in load model code below:
The code works normally as expected when removing the tf.keras.layers.Normalization layer or when I use 1 or 2 GPUs, although when i try to use 4 GPUs the error below occurs. Therefore i discard any problem related to input pipeline, tfrecord parsing, incorrect label format, etc... The error occurs at the end of the epoch.

I created this [gist](https://colab.research.google.com/drive/1-6POGaRSpMhTu5gPO2XC_mtLil7eYXa6?usp=sharing) in order to reproduce the error but everything ran normally before I was able to test other tensorflow versions. I even installed the exact same python version (3.8.10) to verify whether it would be the case but it worked with no errors. 
Back in the original enviroment I've tested limitting the number of GPU devices to 1 or 2 and the code ran normally. Otherwise when i try to run it in 4 GPUs the error occurs.

### Code:

```Python
def load_and_configure_model(optimizer, loss, metrics, path):
  model = ResNet50V2(include_top=True, weights='imagenet')
  transfer_layer = model.get_layer('avg_pool')
  resnet_submodel = Model(inputs=model.input,outputs=transfer_layer.output)
  model_config = resnet_submodel.get_config()
  
  submodel = model_config['layers']
  submodel.remove(submodel[0]) # Remove the previous input layer
  
  input_layer = keras.Input(shape=(224, 224, 3), dtype='float32',name="input") # Create a new input layer
  normalization = tf.keras.layers.Normalization(mean=[118.662, 119.194, 96.877], variance=[2769.232, 2633.742, 2702.492], axis=-1, dtype='float32')(input_layer)
  rescaling = tf.keras.layers.experimental.preprocessing.Rescaling(1./127.5, offset=-1, dtype='float32')(normalization)  
  
  new_model = Model(inputs=input_layer,outputs=rescaling) # Declare pre-processing model to be merged with the ResNet model.
  new_model_cfg = new_model.get_config() 
      
  new_model_cfg['layers'].extend(submodel) # Merge two models.

  # Replace the previous input layer with the output from the preprocessing model
  # (Connect the preprocessing model to the resnet) 
  output_name = new_model_cfg['layers'][2]['name'] # Get the output layer name (rescaling).
  
  new_model_cfg['layers'][3]['inbound_nodes'] = [[[output_name, 0, 0, {}]]] # Replace last inbound node name with the preprocessing model layer name.  
  
  new_model = new_model.__class__.from_config(new_model_cfg, custom_objects={})  # change custom objects if necessary

  # Set back pre-trained weights on new model
  weights = [layer.get_weights() for layer in resnet_submodel.layers[1:]] # For each layer (after the input_layer) in the original resnet50:
  for layer, weight in zip(new_model.layers[3:], weights): # Set imagenet weights on each new model layer.
      layer.set_weights(weight)

  for layer in new_model.layers[:]:
    layer.trainable = False
  for layer in new_model.layers[:]:      
    trainable = True
    layer.trainable = trainable # Train everything. 

  transfer_layer = new_model.get_layer('avg_pool')
  species = Dense(1000, activation='softmax', dtype='float32',name='species')(transfer_layer.output) # Specify dtype for handling mixed precision specifications.

  model = keras.Model(
      inputs=[new_model.inputs],
      outputs=[species],
  )
  if not path == None :
    model.load_weights(path)
  model.compile(optimizer=optimizer, loss=loss, metrics=metrics)
  return model
```

**Comment by gowthamkpr** [comment]
I'm liking the error that you are running into [here](https://discuss.tensorflow.org/t/unable-to-use-multiple-gpus-with-tf-keras-layers-normalization/7663). There are a few things that you can try as mentioned [here](https://stackoverflow.com/questions/60082554/tensorflow-throwing-distributed-function-error). Try all the mentioned solutions and let me know if you are still running into this error. Thanks!
